pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,208 @@
+
+    name : str
+        The name assigned as an attribute to the DataArray.
+
+    long_name : str
+        The long name assigned as an attribute to the data array.
+
+    units : str
+        The units of the values in the DataArray.
+
+    resolutions : dict
+        Dictionary of with given resolution as keys and the values use
+        the Resolution class.
+
+    vertical_datum : str
+        The datum used on the vertical axis.
+
+    horizontal_datum : str
+        The datum used on the horizontal axis.
+    """
+
+    title: str
+    name: str
+    long_name: str
+    units: str
+    resolutions: Dict[str, Resolution]
+    vertical_datum: str
+    horizontal_datum: str
+
+
+datasets = {
+    "earth_relief": Dataset(
+        title="Earth relief",
+        name="elevation",
+        long_name="Earth elevation relative to the geoid",
+        units="meters",
+        vertical_datum="EMG96",
+        horizontal_datum="WGS84",
+        resolutions={
+            "01d": Resolution(["pixel", "gridline"], False),
+            "30m": Resolution(["pixel", "gridline"], False),
+            "20m": Resolution(["pixel", "gridline"], False),
+            "15m": Resolution(["pixel", "gridline"], False),
+            "10m": Resolution(["pixel", "gridline"], False),
+            "06m": Resolution(["pixel", "gridline"], False),
+            "05m": Resolution(["pixel", "gridline"], True),
+            "04m": Resolution(["pixel", "gridline"], True),
+            "03m": Resolution(["pixel", "gridline"], True),
+            "02m": Resolution(["pixel", "gridline"], True),
+            "01m": Resolution(["pixel", "gridline"], True),
+            "30s": Resolution(["pixel", "gridline"], True),
+            "15s": Resolution(["pixel"], True),
+            "03s": Resolution(["gridline"], True),
+            "01s": Resolution(["gridline"], True),
+        },
+    ),
+    "earth_age": Dataset(
+        title="seafloor age",
+        name="seafloor_age",
+        long_name="age of seafloor crust",
+        units="Myr",
+        vertical_datum="EMG96",
+        horizontal_datum="WGS84",
+        resolutions={
+            "01d": Resolution(["pixel", "gridline"], False),
+            "30m": Resolution(["pixel", "gridline"], False),
+            "20m": Resolution(["pixel", "gridline"], False),
+            "15m": Resolution(["pixel", "gridline"], False),
+            "10m": Resolution(["pixel", "gridline"], False),
+            "06m": Resolution(["pixel", "gridline"], False),
+            "05m": Resolution(["pixel", "gridline"], True),
+            "04m": Resolution(["pixel", "gridline"], True),
+            "03m": Resolution(["pixel", "gridline"], True),
+            "02m": Resolution(["pixel", "gridline"], True),
+            "01m": Resolution(["gridline"], True),
+        },
+    ),
+}
+
+
+@kwargs_to_strings(region="sequence")
+def _load_earth_dataset(dataset_name, dataset_prefix, resolution, region, registration):
```

Comment:
Should we also rename `class Dataset` to `class GMTDataset` at L32 then? A little hesitant to have a class with a generic name like `Dataset`, especially since there's [`xarray.Dataset`](https://docs.xarray.dev/en/v2022.12.0/generated/xarray.Dataset.html).

Reply:
> Should we also rename `class Dataset` to `class GMTDataset` at L32 then? A little hesitant to have a class with a generic name like `Dataset`, especially since there's [`xarray.Dataset`](https://docs.xarray.dev/en/v2022.12.0/generated/xarray.Dataset.html).

I agree that `Dataset` is not a good name here. 

I also don't think `GMTDataset` is good because GMT also has a built-in structure called [`GMT_DATASET`](https://github.com/GenericMappingTools/gmt/blob/dfe4e7e31a30b64f1547d8393fa50b572e5ffc81/src/gmt_resources.h#L552). Currently it's not used in PyGMT, but when we wrap this data structure in PyGMT, it's likely will be called `GMTDataset`. 

I think the following two options are better:

1. Rename `Dataset` to `RemoteDataset`/`GMTRemoteDataset`
2. Rename `Dataset` to `_Dataset`/`_RemoteDataset`/`_GMTRemoteDataset` because it won't be used by users.

Reply:
I went with `GMTRemoteDataset`